R/interactive.R

```diff
@@ -1,16 +1,55 @@
+is_plausible_orderly_report <- function(path) {
+  path_split <- fs::path_split(path)[[1]]
+
+  length(path_split) > 2 &&
+    path_split[[length(path_split) - 1]] == "src" &&
+    file.exists(file.path(path, "../..", "orderly_config.yml"))
+}
+
+rstudio_get_current_active_editor_path <- function() {
+  # Avoid looking at the RStudio state when running tests inside of it.
+  if (!is_testing() && rstudioapi::isAvailable()) {
+    rstudioapi::getSourceEditorContext()$path
+  } else {
+    NULL
+  }
+}
+
 ## This is something that we might improve over time - it will likely
 ## be useful to have some sort of "register interactive" function
 ## which we could then just look up.
 ##
 ## I am not sure if we also want to allow working interactively from a
 ## draft directory too.
-detect_orderly_interactive_path <- function(path = getwd()) {
-  path_split <- fs::path_split(path)[[1]]
-  is_plausible <- length(path_split) > 2 &&
-    path_split[[length(path_split) - 1]] == "src" &&
-    file.exists(file.path(path, "../..", "orderly_config.yml"))
-  if (!is_plausible) {
-    stop(sprintf("Failed to detect orderly path at '%s'", path))
+detect_orderly_interactive_path <- function(path = getwd(),
+                                            editor_path = rstudio_get_current_active_editor_path()) {
+  is_valid <- is_plausible_orderly_report(path)
+  suggested_wd <- NULL
+  if (!is.null(editor_path)) {
+    dir <- fs::path_dir(editor_path)
+    if (paths_are_different(dir, path) && is_plausible_orderly_report(dir)) {
+      suggested_wd <- dir
+    }
+  }
+
+  if (!is_plausible_orderly_report(path)) {
+    msg <- c("Working directory {.path {path}} is not a valid orderly report.")
+    if (!is.null(suggested_wd)) {
+      cli::cli_abort(
+        c(msg, i = "Use {.code setwd({.str {suggested_wd}})} to set the working directory to the report currently open in RStudio."))
```

Comment:
You'll need to use `paste()` to fix lint about the line length.

We've not been consistent elsewhere about `{.code ...}` etc and usually just single quote:

```
"Use 'setwd("{suggested_wd}")' to set the working directory to the report currently open in RStudio."
```

I was thinking at some point we'd go through the whole lot. I'm not convinced that showing backticks to the user might not be confusing but at the same time I don't feel strongly.